code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/main.java

```diff
@@ -67,8 +68,9 @@ public static void main(String[] args) throws IOException {
                 StringBuilder stringBuilder = new StringBuilder();
 
                 byte[] buffer = new byte[1024];
-                while (stream.read(buffer) > 0) {
-                    stringBuilder.append(new String(buffer, StandardCharsets.UTF_8));
+                int bytes;
+                while ((bytes = stream.read(buffer)) > 0) {
+                    stringBuilder.append(new String(Arrays.copyOf(buffer, bytes), StandardCharsets.UTF_8));
```

Comment:
Why interpose an `Arrays` object?  Why not just use a better `String` constructor:
```
String(byte[] bytes, int offset, int length, Charset charset)
```

On a related note, does this code actually need to instantiate a new `String`, or can we get away with just passing a subset of `buffer` directly?

Reply:
That's a good callout. I've updated the code to skip the extra buffer copy. 

the bytes need to be converted to string/char somewhere and StringBuffer doesn't expose that functionality.